impl/sql/mutationstorage/dberrors.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mutationstorage
+
+import (
+	"fmt"
+
+	"github.com/VividCortex/mysqlerr"
+	"github.com/go-sql-driver/mysql"
+	"github.com/mattn/go-sqlite3"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// dbErrorf tries to extract a meaningful error code from err,
+// the err returned from the database.
+func dbErrorf(err error, format string, a ...interface{}) error {
+	if err == nil {
+		return nil
+	}
+	msg := fmt.Sprintf(format, a...)
+	if st, ok := status.FromError(err); ok {
+		return status.Errorf(st.Code(), "%v: %v", msg, st.Message())
+	}
+	if sqliteErr, ok := err.(sqlite3.Error); ok {
```

Comment:
I feel slightly opposed to the idea of putting multiple databases handling into one place. Is this list going to grow for every new database?
Should this be hidden somewhere in the DB-specific storage implementation? As well as the retrying logic maybe.

Reply:
Did you observe sqlite failing in a way which would require these restarts?

Also, in that #1363 the core issue seems to be that `main.go` tried to support both DBs. This is the same problem, only on a higher level. Would having two separate `main.go` files for different storages help?

Reply:
I've removed support for multiple databases.